src/main/java/com/revature/responsecollector/JMeterResponseCollector.java

```diff
@@ -0,0 +1,173 @@
+package com.revature.responsecollector;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import org.apache.jmeter.engine.StandardJMeterEngine;
+import org.apache.jmeter.reporters.ResultCollector;
+import org.apache.jmeter.reporters.Summariser;
+import org.apache.jmeter.samplers.SampleEvent;
+import org.apache.jmeter.samplers.SampleResult;
+
+public class JMeterResponseCollector extends ResultCollector { 
+    
+    private static final long serialVersionUID = 1L;
+    private int failCount = 0;
+    private int okResponse = 0;
+    private ArrayList<Long> latencyTimes = new ArrayList<Long>();
+    private long responseMax = 0;
+    
+    private long firstSampleStartTime = 0;
+    private long sampleStartTime;
+    
+    private StandardJMeterEngine engine;
+    private int duration = -1;
+    
+    public JMeterResponseCollector(Summariser summer) {
+        super(summer);
+    }
+    
+    public JMeterResponseCollector(Summariser summer, StandardJMeterEngine engine, int duration) {
+        super(summer);
+        this.engine = engine;
+        this.duration = duration * 1000;
+    }
+    
+
+    @Override
+    public void sampleOccurred(SampleEvent e) {
+        super.sampleOccurred(e);
+        SampleResult r = e.getResult();
+        if (firstSampleStartTime == 0) {
+            firstSampleStartTime = r.getStartTime();
+        }
+        sampleStartTime = r.getStartTime();
+        long latency = r.getLatency();
+        latencyTimes.add(latency);
+        if (latency > responseMax) {
+            responseMax = latency;
+        }
+        if (r.getResponseCode().charAt(0) == '4' || r.getResponseCode().charAt(0) == '5') {
```

Comment:
Probably want a switch case for a count of each category of response codes and do calculation based on that rather than hard coding fail/success counts as 2XX and 4XX/5XX exclusively